easy_phi/app.py

```diff
@@ -387,15 +389,56 @@ def get(self):
 
 
 class AdminConsoleHandler(tornado.web.RequestHandler):
-    """Placeholder for Admin Console web-page"""
+    """ Admin console - system setting configuration """
 
     @auth.http_basic(auth.admin_auth)
     def get(self):
-        self.write("Coming soon..")
+        self.render("admin_home.html")
 
     post = delete = put = get
 
 
+class SystemUpgradeHandler(tornado.web.RequestHandler):
+    """Placeholder for Admin Console web-page"""
+
+    @auth.http_basic(auth.admin_auth)
+    def get(self):
+        self.render("admin_system_upgrade.html",
+                    current_version=__version__,
+                    available_version=utils.get_latest_pypi_version(),
+                    error=None
+                    )
+
+    @auth.http_basic(auth.admin_auth)
+    def post(self):
+        # This method requires certain privileges to install software
+        error = None
+        status = -1
+        try:
+            status = pip.main(['install', '-U', __project__])
+        except OSError:
+            error = "Insufficient permissions to run upgrade. Check system " \
```

Comment:
This part requires system to be installed in user-accessible folder, e.g. using virtualenv. Otherwise, upgrade will require admin privileges and it is not secure